distributed/comm/ucx.py

```diff
@@ -596,14 +612,27 @@ def _scrub_ucx_config():
         if dask.config.get("distributed.comm.ucx.nvlink"):
             tls = tls + ",cuda_ipc"
 
-        options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+        high_level_options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+
+    # Pick up any other ucx environment settings
+    # {"some-name": value} is translated to {"UCX_SOME_NAME": value}
+    def normalise(k):
+        return "_".join(map(str.upper, ("UCX", *k.split("-"))))
+
+    environment_options = {
+        key: v
+        for k, v in dask.config.get("distributed.comm.ucx.environment", {}).items()
+        # High-level settings take precedence over low-level
+        # environment configuration, as do settings already in the
+        # environment (set externally)
+        if (key := normalise(k)) not in os.environ and key[4:] not in high_level_options
```

Comment:
I think logging at `warning` level is a good idea. I don't want to raise an exception of any kind because for debug purposes this might well be useful (e.g. one has defaults in a `yaml` file but temporarily wants to override on the commandline with environment variables).

Reply:
Agreed, warnings should be good.

Reply:
I went for warning when the high level options override the ucx.environment, but only info for os.environ overriding ucx.environment (since this is a more common use case).